commit fed8f2e074a71ff6fcf3d1f04dffb0d4b5279783
Author: Chaoyu <paranoyang@gmail.com>
Date:   Thu Apr 25 14:39:51 2019 -0700

    Validate custom API name (#125)
    
    * validate-custom-api-name
    
    * formatting & lint fix
    
    * fix isidentifier in py2
    
    * Update __init__.py


diff --git a/bentoml/service.py b/bentoml/service.py
index e6f2564..8e5c3eb 100644
--- a/bentoml/service.py
+++ b/bentoml/service.py
@@ -29,6 +29,7 @@ from abc import abstractmethod, ABCMeta
 from bentoml.utils.exceptions import BentoMLException
 from bentoml.service_env import BentoServiceEnv
 from bentoml.artifact import ArtifactCollection
+from bentoml.utils import isidentifier
 from bentoml.utils.s3 import is_s3_url, download_from_s3
 

@@ -153,7 +154,10 @@ def api_decorator(handler_cls, *args, **kwargs):
 
         _set_func_attr(func, '_is_api', True)
         _set_func_attr(func, '_handler', handler)
-        # TODO: validate api_name
+        if not isidentifier(api_name):
+            raise ValueError(
+                "Invalid API name: '{}', a valid identifier must contains only letters, "
+                "numbers, underscores and not starting with a number.".format(api_name))
         _set_func_attr(func, '_api_name', api_name)
 
         return func

